annotation/src/main/java/org/cbioportal/genome_nexus/annotation/service/internal/GeneXrefServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2017 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal Genome Nexus.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.genome_nexus.annotation.service.internal;
+
+import org.cbioportal.genome_nexus.annotation.domain.GeneXref;
+import org.cbioportal.genome_nexus.annotation.service.GeneXrefService;
+import org.cbioportal.genome_nexus.annotation.util.Transformer;
+
+import java.io.IOException;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ *
+ * @author ochoaa
+ */
+@Service
+public class GeneXrefServiceImpl implements GeneXrefService {
+    
+    private String geneXrefsURL;
+    @Value("${genexrefs.url}")
+    public void setGeneXrefsURL(String geneXrefsURL) { this.geneXrefsURL = geneXrefsURL; }
+    
+    @Override
+    public List<GeneXref> getGeneXrefs(String accession) {
+        String xrefJSON = getGeneXrefsJSON(accession);
+        List<GeneXref> geneXrefs = new ArrayList<>();
+        try {
+            geneXrefs = Transformer.mapJsonToInstance(xrefJSON, GeneXref.class);
+        }
+        catch (IOException e) {
+            e.printStackTrace();            
+        }
+        return geneXrefs;
+    }    
+    
+    @Override
+    public String getGeneXrefsJSON(String accession) {
+        String uri = geneXrefsURL.replace("ACCESSION", accession);
+        RestTemplate restTemplate = new RestTemplate();
+        return restTemplate.getForObject(uri, String.class);
```

Comment:
Why not directly marshall the response into a `List<GeneXref>`? Is there a reason to want to get it as a string first? Here is an [example](https://github.com/cBioPortal/genome-nexus-annotation-pipeline/blob/master/annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java#L130).